plugin/core/registry.py

```diff
@@ -41,36 +38,8 @@ def has_manager(self) -> bool:
         return self._manager is not None
 
 
-class LanguageHandlerDispatcher(object):
-
-    def on_start(self, config_name: str, window: WindowLike) -> bool:
-        if config_name in client_start_listeners:
-            return client_start_listeners[config_name](window)
-        else:
-            return True
-
-    def on_initialized(self, config_name: str, window: WindowLike, client: Client) -> None:
-        if config_name in client_initialization_listeners:
-            client_initialization_listeners[config_name](client)
-
-
-def load_handlers() -> None:
-    for handler in LanguageHandler.instantiate_all():
-        register_language_handler(handler)
-    client_configs.update_configs()
-
-
-def register_language_handler(handler: LanguageHandler) -> None:
-    debug("received config {} from {}".format(handler.name, handler.__class__.__name__))
-    client_configs.add_external_config(handler.config)
-    if handler.on_start:
-        client_start_listeners[handler.name] = handler.on_start
-    if handler.on_initialized:
-        client_initialization_listeners[handler.name] = handler.on_initialized
-
-
 def client_from_session(session: Optional[Session]) -> Optional[Client]:
```

Comment:
Maybe correct the return type to be `-> Optional[Type[Client]]` if you plan to keep this function.

One question, you probably plan to remove this function in the future?

Reply:
Do you mean `-> Session` ?

> One question, you probably plan to remove this function in the future?

Yes :)

Reply:
`Optional[Session]` is valid. I would go with that option. 

But `Optional[Type[Client]]` is also valid, because that way we are saying this function will return the base class or any class that extends base class (in our case Session extends Client, that is the reason why it is valid). I have read that in the mypy docs for the `Type` type...

But I would just annotate it with Optional[Session]

Reply:
`Optional[Session]` is valid. I would go with that option. 

But `Optional[Type[Client]]` is also valid, because that way we are saying this function will return the base class or any class that extends base class (in our case Session extends Client, that is the reason why it is valid). I have read that in the mypy docs for the `Type` type...

But I would just annotate it with Optional[Session]